lib/models/command.js

```diff
@@ -197,7 +198,7 @@ let Command = CoreObject.extend({
       return this._currentTask.onInterrupt();
     } else {
       // interrupt all external commands which don't use `runTask()` with an error
-      process.exit(1);
+      return RSVP.reject();
```

Comment:
>  something like RSVP.reject(1)?
> We will process.exit(1) if the cleanup promise (ie the at-exit work) rejects, but the command run promise isn't part of cleanup.

yes. If we do `RSVP.reject(1)` then `capture-exit` just [outputs 1](https://github.com/ember-cli/capture-exit/blob/master/index.js#L97) to the console and exits with `exitCode=1`. As far as I understand this [`catch`](https://github.com/ember-cli/capture-exit/blob/master/index.js#L92) is designed to handle cleanup exceptions. Seems like it's better not to change this behavior. 

Now I think it might be better not to reject on cleanup. We could `resolve` with a desired `exitCode`, something like:
```js
onInterrupt() {
    return RSVP.resolve(1);
    // or even
    return 1;
}
```
Then `capture-exit` can remember the first recognized exitCode and call [`process.exit(firstExitCode)`](https://github.com/ember-cli/capture-exit/blob/master/index.js#L90) after all the interruption handlers are settled.

@hjdivad what do you think? Do you have any objections about `resolve(desiredExitCode)`?